### DISCUSSION TITLE
Another approach to hack the autostart

### DISCUSSION THREAD

**Discussion by Siccity** [nomis/logitech-z906#5] [Show and tell]
This isn't really an issue with the repo, but more a heads-up to those who came here looking for the same things i did.

Instead of hacking the firmware, it is possible to let your z906 start powered ON by using an ATTiny45/85.

Unfortunately i did this a long time ago so I only have limited pictures from the setup, and i do not have the script i used either, so i will try to explain it.

What I did was connect the Attiny 5v power pin to the console power (near the wires somewhere), the ground pin to the ground side of the ON button, and Pin 5 to the other side of the button.

For the programming, all it did was start pin at HIGH, wait a second, set pin 5 to LOW, wait 200ms, and then set it HIGH again. It would do this every time it was powered and then nothing else.

Setting the pin to LOW lets current flow through the chip, effectively acting as if the button had been pressed.

Some of you might read this and think "that's not how you do things" because i am in no way skilled with electricity. But it worked. I've had the ATTiny45 in my Z906 for over a year and it just works. The power button also still works as usual in case you want to power it off again.

I hope this helps someone

![PXL_20220224_220439608](https://user-images.githubusercontent.com/6402525/180599991-c04c3ec3-bd35-4ced-baff-43fc23e0d3cd.jpg)
![PXL_20220224_223006663](https://user-images.githubusercontent.com/6402525/180599994-6a8ff945-46e5-4a8e-9b65-4ac136004e47.jpg)

**Comment by nomis** [comment]
What I'd like to do is find some way to trick the stock console firmware into powering on.

Having determined the purpose of pins 8 and 15 I suspect the console is ignoring the serial communication until the power button is pressed. There's a brief opportunity when power is initially supplied that additional data could be sent to it.

**Comment by Siccity** [comment]
The reason why i did it like this is because i didn't want to mess with the firmware and potentially brick my system. Because messing with that is way over my head

**Comment by nomis** [comment]
As long as you read the existing flash contents first (so you have a backup) and only write to the flash, you can't brick it.

It is possible to enable undesirable bits in the option bytes that can only be cleared by erasing the flash, but that's a separate address from the flash.
